### PR TITLE
Add Hlint rule to disallow the partial `decodeUtf8`.

### PR DESCRIPTION
**Pull request by patrickt**

### CODE REVIEW

File: .hlint.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@
   - {name: init, within: []}
   - {name: last, within: []}
   - {name: fromJust, within: []}
+  - {name: decodeUtf8, within: []}
```
Comment (robrix) [inline .hlint.yaml]: Can we add a `message` suggesting the right way to do this please? 🙏

**Comment by tclem** [comment]
Can we catch usages of [`toText`](https://hackage.haskell.org/package/text-conversions-0.3.0/docs/Data-Text-Conversions.html#v:toText) as well?

**Comment by patrickt** [comment]
@tclem Do we use that package anywhere? I don’t see it in our dependencies.

File: .hlint.yaml [hunk 2]
```diff
@@ -12,6 +12,7 @@
   - {name: init, within: []}
   - {name: last, within: []}
   - {name: fromJust, within: []}
+  - {name: decodeUtf8, within: [], message: "Use decodeUtf8' or decodeUtf8With lenientDecode"}
```
Comment (robrix) [inline .hlint.yaml]: ❤️ ty!

**Pull request merged**